src/apps/service/apps/agave/listings.clj

```diff
@@ -1,24 +1,58 @@
 (ns apps.service.apps.agave.listings
-  (:use [apps.service.util :only [sort-apps apply-offset apply-limit uuid?]]
+  (:use [apps.service.util :only [sort-apps apply-offset apply-limit format-job-stats uuid?]]
+        [apps.util.conversions :only [remove-nil-vals]]
         [slingshot.slingshot :only [try+]])
   (:require [clojure.tools.logging :as log]
             [clojure-commons.error-codes :as ce :refer [clj-http-error?]]
-            [apps.persistence.app-metadata :as ap]))
+            [apps.persistence.app-metadata :as ap]
+            [apps.persistence.jobs :as jobs-db]))
+
+(defn- add-agave-job-stats
+  [{:keys [id] :as app} admin?]
+  (log/spy id)
+  (let [job-stats     (if admin?
+                        (jobs-db/get-job-stats id)
+                        (jobs-db/get-public-job-stats id))
+        default-stats (if admin?
+                        {:job_count 0
+                         :job_count_failed 0
+                         :job_count_completed 0}
+                        {:job_count_completed 0})]
+    (merge app default-stats job-stats)))
```

Comment:
This is not an issue for DE app listings, since the `app_listing` view always includes at least a `0` value for these fields for DE apps that have not run yet.